### PR TITLE
Create Boxel Catalog Realm

### PR DESCRIPTION
**Pull request by FadhlanR**

### CODE REVIEW

File: packages/catalog-realm/TODO.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Make sure to lint for hbs when we add gts files to this realm like we do in the experiments realm. Currently linting for templates fails because there are no gts files in this realm.
```
Comment (tintinthong) [inline packages/catalog-realm/TODO.md:1]: Can't tell if this is meant to be a real file or your notes to self

**Comment by tintinthong** [comment]
Screenshot 2024-09-18 at 22 01 39

For this PR, is this the only way to access the catalog?

File: packages/catalog-realm/TODO.md (hunk 1 above)
Comment (habdelra) [reply packages/catalog-realm/TODO.md:1]: it's a real TODO. as soon as we actually put gts files in in this realm (which we will), we need to turn on linting. we can remove this TODO after that.

Comment (habdelra) [reply packages/catalog-realm/TODO.md:1]: it would suck to forget to lint this realm...

**Comment by habdelra** [comment]
> Screenshot 2024-09-18 at 22 01 39
> For this PR, is this the only way to access the catalog?

currently yes, but very shortly we'll be adding this as a realm in the user's dashboard. it will be in a followup PR (either me or fadhlan will take care of it)

**Comment by habdelra** [comment]
This looks good. you will also need to create a catalog_realm user in matrix on both staging and production using the secret seed before you deploy this. The best was to do that is to use the matrix admin console to make a catalog_realm user with any value as the password, then run the migrate realm user script to update the user to have a seed based password.

Here are examples that can be run in the matrix workspace, where you can fill in the xxx with values from AWS parameter store

staging:
```
ADMIN_USERNAME='xxx' ADMIN_PASSWORD='xxx' MATRIX_URL='https://matrix-staging.stack.cards' REALM_SECRET_SEED='xxx' ts-node --transpileOnly ./scripts/migrate-realm-user '@catalog_realm:stack.cards'
```

production:
```
ADMIN_USERNAME='xxx' ADMIN_PASSWORD='xxx' MATRIX_URL='https://matrix.boxel.ai' REALM_SECRET_SEED='xxx' ts-node --transpileOnly ./scripts/migrate-realm-user '@catalog_realm:boxel.ai'
```

**Review by habdelra — APPROVED** [review]

File: packages/realm-server/main.ts [hunk 2]
```diff
@@ -343,7 +343,10 @@ async function startWorker() {
 // Other realms: read permissions for everyone, read/write permissions for signed up users
 async function seedRealmPermissions(dbAdapter: PgAdapter, realmURL: URL) {
   if (!(await permissionsExist(dbAdapter, realmURL))) {
-    if (realmURL.href === 'https://cardstack.com/base/') {
+    if (
+      realmURL.href === 'https://cardstack.com/base/' ||
+      realmURL.href.split('/')[1] === 'catalog'
```
Comment (tintinthong) [inline packages/realm-server/main.ts]: Is this right? Or .pathname.split('/')

Comment (habdelra) [reply packages/realm-server/main.ts]: Actually I have set this as part of the migration scripts in my PR. no need to set this for base or catalog realm https://github.com/cardstack/boxel/pull/1582. as soon as this lands the permissions should be set correctly for these 2 realms (all of them actually)

**Review by tintinthong — APPROVED** [review]

**Pull request merged**